File: 2022-2023/sourcecode/lab/lab11/VeryNobleAppJava/app/src/main/java/pl/udu/uwr/pum/verynobleappjava/api/NobelPrizeApi.java
package pl.udu.uwr.pum.verynobleappjava.api;

import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.LaureateResponse;
import pl.udu.uwr.pum.verynobleappjava.data.nobelprizeresponse.NobelAwardsResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface NobelPrizeApi {
    @GET("2.1/nobelPrizes")
    Call<NobelAwardsResponse> getNobelPrizes(
            @Query("limit") int limit,
            @Query("sort") String sort,
            @Query("nobelPrizeYear") int yearFrom,
            @Query("yearTo") int yearTo,
            @Query("nobelPrizeCategory") String category
    );

    @GET("2.1/nobelPrizes")
    Call<NobelAwardsResponse> getNobelPrize(
            @Query("nobelPrizeYear") int year,
            @Query("nobelPrizeCategory") String category
    );

    @GET("2.1/laureate/{laureateID}")
    Call<LaureateResponse> getLaureates(@Path("laureateID") String id);
}

File: 2022-2023/sourcecode/lab/lab11/VeryNobleAppKotlin/app/src/main/java/pl/udu/uwr/pum/verynobleappkotlin/api/NobelPrizeApi.kt
package pl.udu.uwr.pum.verynobleappkotlin.api

import pl.udu.uwr.pum.verynobleappkotlin.data.nobelprizeresponse.NobelPrizeResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query


interface NobelPrizeApi {
    @GET("2.1/nobelPrizes")
    suspend fun getNobelPrizes(
        @Query("limit") limit: Int = 200,
        @Query("sort") sort: String = "desc",
        @Query("nobelPrizeYear") yearFrom: Int = 1901,
        @Query("yearTo") yearTo: Int = 2022,
        @Query("nobelPrizeCategory") category: String,
        @Query("format") format: String = "json"
    ) : Response<NobelPrizeResponse>
}